tools/StaticAnalysis/BreakingChangeAnalyzer/BreakingChangeAnalyzer.cs

```diff
@@ -0,0 +1,968 @@
+                                remediation: string.Format(Properties.Resources.RemovedParameterSetRemediation, 
+                                    oldParameterSet.Name, oldCmdlet.Name));
+                        }
+                    }
+                }
+                // If the cmdlet is not in the new assembly, log an issue
+                else
+                {
+                    issueLogger.LogBreakingChangeIssue(
+                        cmdlet: oldCmdlet,
+                        severity: 0,
+                        problemId: ProblemIds.BreakingChangeProblemId.RemovedCmdlet,
+                        description: string.Format(Properties.Resources.RemovedCmdletDescription, oldCmdlet.Name),
+                        remediation: string.Format(Properties.Resources.RemovedCmdletRemediation, oldCmdlet.Name));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Run all of the different breaking change checks for parameters
+        /// </summary>
+        /// <param name="cmdlet"></param>
+        /// <param name="oldParameter"></param>
+        /// <param name="newParameter"></param>
+        /// <param name="issueLogger"></param>
+        private void CheckParameters(
+            CmdletBreakingChangeMetadata cmdlet,
+            ParameterMetadata oldParameter,
+            ParameterMetadata newParameter,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            CheckForChangedParameterType(cmdlet, oldParameter, newParameter, issueLogger);
+            CheckForRemovedParameterAlias(cmdlet, oldParameter, newParameter, issueLogger);
+            CheckParameterValidationSets(cmdlet, oldParameter, newParameter, issueLogger);
+            CheckForValidateNotNullOrEmpty(cmdlet, oldParameter, newParameter, issueLogger);
+        }
+
+        /// <summary>
+        /// Run all of the different breaking change checks for parameter sets
+        /// </summary>
+        /// <param name="cmdlet"></param>
+        /// <param name="oldParameterSet"></param>
+        /// <param name="newParameterSet"></param>
+        /// <param name="issueLogger"></param>
+        private void CheckParameterSets(
+            CmdletBreakingChangeMetadata cmdlet,
+            ParameterSetMetadata oldParameterSet,
+            ParameterSetMetadata newParameterSet,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            // For each of the parameters in the old parameter set assembly, find the parameter
+            // in the new parameter set assembly that either has the same name, or an alias to
+            // the original name
+            foreach (var oldParameter in oldParameterSet.Parameters)
+            {
+                bool found = false;
+
+                foreach (var newParameter in newParameterSet.Parameters)
+                {
+                    if (oldParameter.ParameterMetadata.Name.Equals(newParameter.ParameterMetadata.Name) ||
```

Comment:
I think we should be explicit about comparison, we need to use StringComparison.OrdinalIgnoreCase when comparing strings.